xbmc/PlayListPlayer.cpp

```diff
@@ -298,7 +298,12 @@ bool CPlayListPlayer::Play(int iSong, std::string player, bool bAutoPlay /* = fa
   m_iCurrentSong = iSong;
   CFileItemPtr item = playlist[m_iCurrentSong];
   if (item->IsVideoDb() && !item->HasVideoInfoTag())
-    *(item->GetVideoInfoTag()) = XFILE::CVideoDatabaseFile::GetVideoTag(CURL(item->GetPath()));
+  {
+    auto path = !item->GetDynPath().empty() ? item->GetDynPath() : item->GetPath();
```

Comment:
I think GetDynPath have already an fallback if GetDynPath is empty 
https://github.com/xbmc/xbmc/blob/2f43cc1bbde193869ee11c690d14de82fc6913eb/xbmc/FileItem.cpp#L1758-L1765

Reply:
That's not the goal here.
The line below will fill the `infotag` of the current item with the `infotag` of the `dynpath` if it is set otherwise it will just use the `path`. Thus, if you have a `.strm` file containing a `videodb://` url, the variables will be as follows:

- path = .strm file
- dynpath = `videodb://` url

This method will fill the video `infotag` of the `ListItem` with the video `infotag` of `videodb://` (we don't want the .strm) and replace the `dynpath` with the `m_strFileNameAndPath` saved on the db - allowing the playback.

Reply:
I thought the two things were doing the same thing, if not sorry for the noise.

```suggestion
    auto path = !item->GetDynPath().empty() ? item->GetDynPath() : item->GetPath();
```
```suggestion
    auto path = item->GetDynPath();
```

Reply:
you're right, thanks. Fixed.